src/components/popups.vue

```diff
@@ -212,152 +244,130 @@ import { Apis } from "bitsharesjs-ws";
-                this.genericaccept=this.$t('operations:vote.accept_btn');
-                this.genericreject=this.$t('operations:vote.reject_btn');
+                this.genericmsg = this.$t(
+                    'operations:vote.request',
+                    {
+                        origin: this.incoming.origin,
+                        entity: mappedData.entity
+                    }
+                );
+                this.generictitle = this.$t('operations:vote.title');
+                this.genericaccept = this.$t('operations:vote.accept_btn');
+                this.genericreject = this.$t('operations:vote.reject_btn');
                 this.$refs.genericReqModal.show();
                 return new Promise((res, rej) => {
                     this.incoming.acceptgen = res;
                     this.incoming.rejectgen = rej;
                 });
             },
-            requestTx: function(request) {
+            requestTx: function (payload) {
                 this.$store.dispatch("WalletStore/notifyUser", {
                     notify: "request", message: "request"
                 });
-                this.incoming = {};
-                this.incoming = request;
+                this.incoming = payload;
                 this.$refs.transactionReqModal.show();
                 return new Promise((res, rej) => {
                     this.incoming.accepttx = res;
                     this.incoming.rejecttx = rej;
                 });
             },
-            allowAccess: function() {
+            signMessage: function (payload) {
+                let blockchain = getBlockchain(this.$store.state.WalletStore.wallet.chain);
+                return new Promise((resolve, reject) => {
+                    blockchain.signMessage(
+                        this.$store.state.WalletStore.wallet.keys.active,
+                        this.$store.state.WalletStore.wallet.accountName,
+                        payload.params
+                    ).then((signedMessage) => {
+                        blockchain.verifyMessage(signedMessage).then(res => {
+                            resolve(signedMessage);
+                        }).catch(err => {
+                            reject(err);
+                        });
+                    }).catch(err => {
+                        reject(err);
+                    });
+                });
+            },
+            verifyMessage: function (payload) {
+                console.log("verify", payload);
+                let blockchain = getBlockchain(payload.params.payload[2].substring(0,3));
+                return new Promise((resolve, reject) => {
+                    blockchain.verifyMessage(
+                        payload.params
+                    ).then((result) => {
+                        resolve(result);
+                    }).catch(err => {
+                        reject(err);
+                    });
+                });
+            },
```

Comment:
See: https://github.com/bitshares/beet/blob/Steem-Support/src/lib/BeetWS.js#L86

Inject chain type to the data object sent to the API. Will make things much cleaner

Reply:
You mean as a separate field in the signed message definition?

Reply:
Misunderstood that part of your code. This verifies against the OTHER user's chain right?

Looked through the code again. Why do you pass the public key through? Incoming message should basically be account-name, chain-identifier, message, signature. 

Verify looks up keys on the indicated chain for account-name and verifies the signature.

Reply:
Yes exactly, it verifies messages from any of the supported chains.

The public key is added as the chain identifier (first three digits), and I decided to add the whole public key since that facilitates the verification and does not hurt a lot in terms of payload.